### PR TITLE
kaitai-struct-compiler: init at 0.9

### PR DESCRIPTION
**Pull request by Qubasa**
###### Motivation for this change
Packaged the compiler for [kaitai struct ](https://doc.kaitai.io/)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/kaitai_struct/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  installPhase = ''
+    install -D $src/bin/kaitai-struct-compiler $out/bin/ksc
+    cp -R $src/lib $out/lib
+    wrapProgram $out/bin/ksc --prefix PATH : ${lib.makeBinPath [ openjdk8 ] }
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/kaitai-io/kaitai_struct_compiler";
+    description =
+      "Kaitai Struct: declarative language to generate binary data parsers in C++ / C# / Go / Java / JavaScript / Lua / Perl / PHP / Python / Ruby ";
```
Comment (ius) [inline pkgs/development/compilers/kaitai_struct/default.nix:30]: According to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/030ff68d47fb8595b46f56f58955a55fe663303f/CONTRIBUTING.md?plain=1#L35) the description should not start with the package name

Should also probably mention that it's just the compiler (?)

File: pkgs/development/compilers/kaitai_struct/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+, stdenv
+, fetchzip
+, sbt-with-scala-native
+, openjdk8
+, makeWrapper
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "kaitai_struct_compiler";
```
Comment (ius) [inline pkgs/development/compilers/kaitai_struct/default.nix:11]: Shouldn't this be hyphenated instead?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -22270,6 +22270,8 @@ with pkgs;
 
   lsiutil = callPackage ../os-specific/linux/lsiutil { };
 
+  kaitai_struct = callPackage ../development/compilers/kaitai_struct { };
```
Comment (ius) [inline pkgs/top-level/all-packages.nix:22273]: A `kaitaistruct` attribute already exists... && same hyphen remark as `pname`

File: pkgs/development/compilers/kaitai_struct/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchzip
+, sbt-with-scala-native
```
Comment (ius) [inline pkgs/development/compilers/kaitai_struct/default.nix:4]: Not used?

**Review by ius — CHANGES_REQUESTED** [review]
~~Shouldn't this be built from source?~~

**Comment by Mic92** [comment]
> Shouldn't this be built from source?

Dependence on the state of sbt based tooling for nix. Do you have experience with building scala from source in nix?

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/top-level/all-packages.nix:22273]: This is how other projects named it: https://repology.org/project/kaitai-struct/versions

**Comment by ius** [comment]
> > Shouldn't this be built from source?
> 
> Dependence on the state of sbt based tooling for nix. Do you have experience with building scala from source in nix? My impression with java-based packaging is that one does not gain much from a technical perspective.

Upon closer inspection, it seems quite limited indeed. `sbt` is packaged, but running `sbt compile` will attempt to download dependencies.

I guess it's fine then (and I suppose you're right on the second bit as well).

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -22270,6 +22270,8 @@ with pkgs;
 
   lsiutil = callPackage ../os-specific/linux/lsiutil { };
 
+  kaitai-struct-compiler = callPackage ../development/compilers/kaitai_struct { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:22273]:
```suggestion
  kaitai-struct-compiler = callPackage ../development/compilers/kaita-struct-compiler { };
```

**Pull request merged**